astropy/coordinates/baseframe.py

```diff
@@ -454,10 +454,63 @@ def representation_component_units(self):
                 out[repr_name] = repr_unit
         return out
 
+    def replicate(self, copy=False, **kwargs):
+        """
+        Generates a new frame object that is a copy of this one, but possibly
```

Comment:
It is mostly a matter of taste, but I prefer single-line "titles" of the docstring. E.g., inspired by `Time`s:
```
Return a replicate of the frame, optionally overriding some attributes.
```